lesson_22/test_api.py

```diff
@@ -1,63 +1,341 @@
+    email, password = create_user_data['email'], create_user_data["password"]
 
+    # Login(Signin)
+    login_response = login(s, email, password)
+    assert login_response.status_code == 200, f"Wrong status code:\n{login_response.json()}"
+    assert login_response.json().get('status', False) == "ok", "Key 'status' is not ok, User wasn't logged in"
 
-def test_signin_positive():
-    user_data = {
-        "email": "qam2808@2022test.com",
-        "password": "Qam2608venv",
-        "remember": False
+    # Reset password
+    reset_password_data = {
+        "email": email,
     }
-    r = auth.signin(s, user_data)
-    r_json = r.json()
-    assert r.status_code == 200, f"Wrong status code:\n{r_json}"
-    assert r_json.get('status', False) == "ok", "Key 'status' is not ok"
+    reset_password_response = auth.resetpassword(s, reset_password_data)
+    assert reset_password_response.status_code == 200, f"Wrong status code:\n{reset_password_response.json()}"
+    assert reset_password_response.json().get('status', False) == "ok", \
+        "Key 'status' is not ok, Reset password email wasn't sent"
 
-def get_a(a):
-    return int(a)
+    # Logout
+    logout_response = auth.logout(s)
+    assert logout_response.status_code == 200, f"Wrong status code:\n{logout_response.json()}"
+    assert logout_response.json().get('status', False) == "ok", "Key 'status' is not ok. User wasn't logged out"
 
-def test_error():
-    with pytest.raises(
-         ValueError
-         ):
-        a ='dsdsd'
-        get_a(a)
 
+"""
+Test 2:
+Get User Profile info
+Update User Profile info
+Get User Profile settings
+Update User Profile settings:
+1) Try to change currency that not allowed
+2) Update currency and distanceUnits and assert that changes were applied
+"""
 
-def get_error():
-    raise AttributeError("character name empty")
 
+def test_UserProfileFlow(login):
+    # Login
+    user_login = login(s, email, password)
+
+    # Get User Profile info
+    user_profile_info = users.profile_get(s)
+    assert user_profile_info.status_code == 200, f"Wrong status code:\n{user_profile_info.json()}"
+    for key in ('userId', 'photoFilename', 'name', 'lastName'):
+        assert key in user_profile_info.json()['data'], f"Key {key} is not in data"
+
+    # Update User Profile info
+    user_update_data = {
+        "photo": 'default-user.png',
+        "name": random.choice(First_Name),
+        "lastName": random.choice(Last_Name),
+        "dateBirth": "2021-03-17T15:21:05.000Z",
+        "country": "Ukraine"
+    }
+    user_update_info = users.profile_put(s, user_update_data)
+    assert user_update_info.status_code == 200, f"Wrong status code:\n{user_update_info.json()}"
+    assert user_update_info.json().get('status', False) == "ok", "Key 'status' is not ok"
+    for key in user_update_data.keys():
+        if key == 'photo':
+            assert user_update_data[key] == user_update_info.json()['data']['photoFilename'], f"Key {key} is not equal"
```

Comment:
тут добре, але можна було і так:

```suggestion
            keyword =  'photoFilename' if key == 'photo' else key
            assert user_update_data[keyword ] == user_update_info.json()['data'][keyword ], f"Key {keyword } is not equal"
```